src/raster/rasterband.rs

```diff
@@ -852,38 +880,230 @@ pub struct CmykEntry {
 ///
-/// This object carries the lifetime of the raster band that
-/// contains it. This is necessary to prevent the raster band
-/// from being dropped before the color table.
+/// // Open source multinomial classification raster
+/// let ds = Dataset::open("fixtures/labels.tif")?;
+///
+/// // Create in-memory copy to mutate
+/// let mem_driver = Driver::get_by_name("MEM")?;
+/// let ds = ds.create_copy(&mem_driver, "<mem>", &[])?;
+/// let mut band = ds.rasterband(1)?;
+/// assert!(band.color_table().is_none());
+///
+/// // Create a new color table for 3 classes + no-data
+/// let mut ct = ColorTable::new(PaletteInterpretation::Rgba);
+/// ct.set_color_entry(0, &ColorEntry::rgba(255, 255, 0, 255));
+/// ct.set_color_entry(1, &ColorEntry::rgba(0, 255, 255, 255));
+/// ct.set_color_entry(2, &ColorEntry::rgba(255, 0, 255, 255));
+/// ct.set_color_entry(255, &ColorEntry::rgba(0, 0, 0, 0));
+/// band.set_color_table(&ct);
+///
+/// // Render a PNG
+/// let png_driver = Driver::get_by_name("PNG")?;
+/// ds.create_copy(&png_driver, "/tmp/labels.png", &[])?;
+///
+/// # Ok(())
+/// # }
+/// ```
 pub struct ColorTable<'a> {
     palette_interpretation: PaletteInterpretation,
     c_color_table: GDALColorTableH,
+    /// If `true`, Rust is responsible for deallocating color table pointed to by
+    /// `c_color_table`, which is the case when instantiated directly, as opposed to
+    /// when read via [`RasterBand::color_table`].
+    rust_owned: bool,
     phantom_raster_band: PhantomData<&'a RasterBand<'a>>,
 }
 
 impl<'a> ColorTable<'a> {
+    /// Instantiate a new color table with the given palette interpretation.
+    pub fn new(interp: PaletteInterpretation) -> Self {
+        let c_color_table = unsafe { GDALCreateColorTable(interp.c_int()) };
+        Self {
+            palette_interpretation: interp,
+            c_color_table,
+            rust_owned: true,
+            phantom_raster_band: PhantomData,
+        }
+    }
+
+    /// Constructs a color ramp from one color entry to another.
+    ///
+    /// `start_index` and `end_index` must be `0..=255`.
+    ///
+    /// Returns `None` if `start_color` and `end_color` do not have the same [`PaletteInterpretation`].
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::{Dataset, Driver};
+    /// use gdal::raster::{ColorEntry, ColorTable, PaletteInterpretation};
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// // Create a 16 step blue to white color table.
+    /// let ct = ColorTable::color_ramp(
+    ///     0, &ColorEntry::rgba(0, 0, 255, 255),
+    ///     15, &ColorEntry::rgba(255, 255, 255, 255)
+    /// )?;
+    /// println!("{ct:?}");
+    /// # Ok(())
+    /// # }
+    pub fn color_ramp(
+        start_index: u8,
+        start_color: &ColorEntry,
+        end_index: u8,
+        end_color: &ColorEntry,
+    ) -> Result<ColorTable<'a>> {
+        if start_color.palette_interpretation() != end_color.palette_interpretation() {
```

Comment:
I might be wrong, but it feels like this might just as well be an `assert_eq!`, since it's a user error to pass in different types of entries.

But it's fine to leave it like this, if there's prior art or you think it's more user-friendly.

Reply:
I lean toward *not* `panic!`-ing on behalf of the user whenever possible. If they want to `unwrap` and get a `panic!` that's their business, but rather give them the option to handle it more gracefully if they want.

Reply:
What I would have rather done is make it a compile error :) But that's a rabbit hole.